app/models/invite.rb

```diff
@@ -40,18 +51,30 @@ def user_doesnt_already_exist
     return if email.blank?
     user = Invite.find_user_by_email(email)
 
-    if user && user.id != self.user_id
+    if user && user.id != self&.user&.user_id
```

Comment:
You're right about 1.

I think I wanted this to be `self.invited_users&.first&.user_id`. Because `Invite` model doesn't have direct relation to user.